src/test/java/com/powsybl/openloadflow/OpenLoadFlowParametersTest.java

```diff
@@ -69,22 +77,70 @@ public void testConfig() {
     }
+
+    @Test
+    public void testInvalidOpenLoadflowConfig() {
+        MapModuleConfig olfModuleConfig = platformConfig.createModuleConfig("open-loadflow-default-parameters");
+        // Invalid -> SlackBusSelectorParametersReader cannot be found
+        olfModuleConfig.setStringProperty("slackBusSelectorType", "Invalid");
+
+        LoadFlowParameters parameters = LoadFlowParameters.load(platformConfig);
+        OpenLoadFlowParameters olfParameters = parameters.getExtension(OpenLoadFlowParameters.class);
+
+        // Default value are selected and error log message is printed
+        assertEquals(SLACK_BUS_SELECTOR_DEFAULT_VALUE, olfParameters.getSlackBusSelector());
+    }
+
+    @Test
+    public void testFirstSlackBusSelector() throws IOException {
+        Path cfgDir = Files.createDirectory(fileSystem.getPath("config"));
+        Path cfgFile = cfgDir.resolve("configFirstSlackBusSelector.yml");
+
+        Files.copy(getClass().getResourceAsStream("/configFirstSlackBusSelector.yml"), cfgFile);
+        PlatformConfig platformConfig = new PlatformConfig(new YamlModuleConfigRepository(cfgFile), cfgDir);
+
+        LoadFlowParameters parameters = LoadFlowParameters.load(platformConfig);
+        OpenLoadFlowParameters olfParameters = parameters.getExtension(OpenLoadFlowParameters.class);
+        assertEquals(FirstSlackBusSelector.class, olfParameters.getSlackBusSelector().getClass());
+    }
+
+    @Test
+    public void testMostMeshedSlackBusSelector() throws IOException {
+        Path cfgDir = Files.createDirectory(fileSystem.getPath("config"));
+        Path cfgFile = cfgDir.resolve("configMostMeshedSlackBusSelector.yml");
+
+        Files.copy(getClass().getResourceAsStream("/configMostMeshedSlackBusSelector.yml"), cfgFile);
+        PlatformConfig platformConfig = new PlatformConfig(new YamlModuleConfigRepository(cfgFile), cfgDir);
+
+        LoadFlowParameters parameters = LoadFlowParameters.load(platformConfig);
+        OpenLoadFlowParameters olfParameters = parameters.getExtension(OpenLoadFlowParameters.class);
+        assertEquals(MostMeshedSlackBusSelector.class, olfParameters.getSlackBusSelector().getClass());
+    }
+
+    @Test
+    public void testNameMeshedSlackBusSelector() throws IOException {
+        Path cfgDir = Files.createDirectory(fileSystem.getPath("config"));
+        Path cfgFile = cfgDir.resolve("configNameSlackBusSelector.yml");
+
+        Files.copy(getClass().getResourceAsStream("/configNameSlackBusSelector.yml"), cfgFile);
+        PlatformConfig platformConfig = new PlatformConfig(new YamlModuleConfigRepository(cfgFile), cfgDir);
+
+        LoadFlowParameters parameters = LoadFlowParameters.load(platformConfig);
+        OpenLoadFlowParameters olfParameters = parameters.getExtension(OpenLoadFlowParameters.class);
+        assertEquals(NameSlackBusSelector.class, olfParameters.getSlackBusSelector().getClass());
+    }
```

Comment:
Could you just add an assert to check if the busId is read correctly?

Reply:
Good catch

Reply:
I would have tested an existing bus (like VLHV1_0 for instance) instead of testing the exception raised by an unknown bus, but that's ok!